src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -0,0 +1,451 @@
+#include "parseSubscription.h"
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include <algorithm>
+
+#include "rapidjson/document.h"
+
+#include "common/globals.h"
+#include "rest/ConnectionInfo.h"
+#include "rest/OrionError.h"
+#include "ngsi/ParseData.h"
+#include "ngsi/Request.h"
+#include "jsonParseV2/jsonParseTypeNames.h"
+
+#include "jsonParseV2/parseSubscription.h"
+
+using namespace rapidjson;
+
+
+/* Prototypes */
+static std::string parseAttributeList(ConnectionInfo* ciP, std::vector<std::string>* vec, const Value& attributes);
+static std::string parseNotification(ConnectionInfo* ciP, SubscribeContextRequest* scrP, const Value& notification);
+static std::string parseSubject(ConnectionInfo* ciP, SubscribeContextRequest* scrP, const Value& subject);
+static std::string parseEntitiesVector(ConnectionInfo* ciP, EntityIdVector* eivP, const Value& entities);
+static std::string parseNotifyConditionVector(ConnectionInfo* ciP, NotifyConditionVector* ncvP, const Value& condition);
+
+
+/* ****************************************************************************
+*
+* parseSubscription -
+*
+*/
+std::string parseSubscription(ConnectionInfo* ciP, ParseData* parseDataP)
+{
+  Document    document;
+  std::string r;
+
+  document.Parse(ciP->payload);
+
+  if (document.HasParseError())
+  {
+    LM_W(("Bad Input (JSON parse error)"));
+    OrionError oe(SccBadRequest, "Errors found in incoming JSON buffer");
+
+    return oe.render(ciP, "");
+  }
+
+  if (!document.IsObject())
+  {
+    LM_E(("Bad Input (JSON Parse Error)"));
```

Comment:
LM_W for bad input, however, alarmMgr should be used now:

    #include "alarmMgr/alarmMgr.h"

    alarmMgr.badInput(clientIp, "JSON Parse Error");